src/ir/mod.rs

```diff
@@ -15,39 +18,48 @@
 *
 * SPDX-License-Identifier: Apache-2.0
 */
-
 use std::collections::HashMap;
 
 use chrono::{DateTime, Utc};
 use influx_db_client::keys::{Point, Value};
 
-/// An intermediate representation of the data that
-/// isn't as lossy as influx_db_client's Point enum
+/// An intermediate representation of time series data points
 #[derive(Clone, Debug)]
 pub struct TsPoint {
     pub measurement: String,
     pub tags: HashMap<String, TsValue>,
     pub fields: HashMap<String, TsValue>,
+    /// This field is generally used for indexing
     pub timestamp: Option<DateTime<Utc>>,
+    /// Optionally specify a field that should be used for indexing values.
+    /// If not specified then the timestamp field will be used.
+    pub index_field: Option<String>,
 }
 
 impl TsPoint {
-    pub fn new(measurement: &str) -> TsPoint {
+    pub fn new(measurement: &str, is_time_series: bool) -> TsPoint {
         TsPoint {
             measurement: String::from(measurement),
             tags: HashMap::new(),
             fields: HashMap::new(),
-            timestamp: Some(Utc::now()),
+            timestamp: if is_time_series { Some(Utc::now())} else {None},
+            index_field: None,
         }
     }
+
+    /// Add a field and its value
+    pub fn add_field<T: ToString>(&mut self, field: T, value: TsValue) {
+        self.fields.insert(field.to_string(), value);
+    }
+
     /// Add a tag and its value
     pub fn add_tag<T: ToString>(&mut self, tag: T, value: TsValue) {
         self.tags.insert(tag.to_string(), value);
     }
 
-    /// Add a field and its value
-    pub fn add_field<T: ToString>(&mut self, field: T, value: TsValue) {
-        self.fields.insert(field.to_string(), value);
+    /// Set the field to be used for indexing if supported
+    pub fn set_index_field<T: ToString>(&mut self, f: T)  {
```

Comment:
Oh I should change this to a better variable name.  Since the fields are all "key":TsValue mappings i made this to allow the library user to set whatever string field they want as the index field.

Reply:
Yeah you're right.  This will have consequences.  It's going to break the isilon and scaleio conversions in addition to picking up on a lot of other fields it was ignoring before.